File: 0283-move-zeroes/solution.java
class Solution {
    public void moveZeroes(int[] nums) {
     int  end= nums.length;
        for(int i=0;i<end;i++){
            if(nums[i]==0){
                leftRotate(nums,i,end);
                
            if(nums[i]==0)
                i--;
                
                end--;
            }
        }
    }
    void  leftRotate(int nums[],int start,int end){
      for(int i=start;i<end-1;i++)
          nums[i]=nums[i+1];
        
        nums[end-1]=0;
    } 
}

File: 0283-move-zeroes/solution.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int k = -1;
        // first find 1st zero
        for(int i=0; i<nums.size(); i++){
            if(nums[i] == 0){
                k = i;
                break;
            }
        }
        if(k == -1) return;

        for(int i=k+1; i<nums.size(); i++){
            if(nums[i] != 0){
                swap(nums[i], nums[k]);
                k++;
            }
        }
    }
};
